### PR TITLE
Add classes for LID/VID/LIDVID and fix #224

### LINKED ISSUE #224: registry-api does not return latest version of product metadata when multiple versions are harvested
## 🐛 Describe the bug
Given that v7.0 of the example product is harvested and set to archived status, if v14.0 is harvested and ingested:
- Requesting the LID returns v7.0 metadata
- Requesting the v14.0 LIDVID succeeds, but returns v7.0 metadata.

## 📜 To Reproduce
1. Deploy docker-composed registry `int-registry-batch-loader` profile (which includes `urn:nasa:pds:insight_rad:data_derived::7.0`
2. Run `harvest` against the latest version of `urn:nasa:pds:insight_rad` bundle
3. Run `registry-mgr` against the LIDVID of the bundle or `data_derived` collection
4. Run `curl --get 'http://localhost:8080/collections/urn:nasa:pds:insight_rad:data_derived' | json_pp`, observe v7.0 metadata returned
5. Run `curl --get 'http://localhost:8080/collections/urn:nasa:pds:insight_rad:data_derived::14.0' | json_pp` (i.e latest version LIDVID), observe v7.0 metadata returned

## 🕵️ Expected behavior
Expect return of v14.0 metadata

## 📚 Version of Software Used
Docker image `nasapds/registry-api-service:1.1.11`

### PR DESCRIPTION
**Pull request by alexdunnjpl**
## 🗒️ Summary
Introduces LID/VID/LIDVID as first-class objects, including comparison operations.

## ⚙️ Test Data and/or Report
Unit tests added for new classes, manually-tested test cases for #224 

## ♻️ Related Issues
fixes #224
causes #234 as a side-effect of fixing #224 (but I'll get to that)

### CODE REVIEW

**Comment by alexdunnjpl** [comment]
Sorry for all the whitespace changes - do we have an autoformat configuration for our projects kicking around?  I'm just using the IDEA defaults currently.

File: service/src/main/java/gov/nasa/pds/api/registry/model/PdsLid.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package gov.nasa.pds.api.registry.model;
```
Comment (nutjob4life) [inline service/src/main/java/gov/nasa/pds/api/registry/model/PdsLid.java:1]: Kind of surprised we don't already have a "data class" like this, perhaps in `pds4-jparser`. Oh well, better late than never!

**Review by nutjob4life — APPROVED** [review]
LGTM

File: service/src/main/java/gov/nasa/pds/api/registry/model/PdsLid.java (hunk 1 above)
Comment (alexdunnjpl) [reply service/src/main/java/gov/nasa/pds/api/registry/model/PdsLid.java:1]: We may do, but if so it's not in use in _this_ repo.  Is it worth me following that up, or am I good to continue with this for this repo, at least?

Comment (alexdunnjpl) [reply service/src/main/java/gov/nasa/pds/api/registry/model/PdsLid.java:1]: You know what?  After looking at it I think that module might be _slight_ overkill for the task of sorting VIDs (actually I don't even think it's capable of that, but it *is* very imposing and impressive)

**Pull request merged**